src/librustc_driver/lib.rs

```diff
@@ -593,6 +596,32 @@ fn show_content_with_pager(content: &String) {
 }
 
 impl RustcDefaultCalls {
+    fn process_rlink(sess: &Session, compiler: &interface::Compiler) -> Result<(), ErrorReported> {
+        if let Input::File(file) = compiler.input() {
+            let attrs = vec![];
+            sess.crate_types.set(collect_crate_types(sess, &attrs));
```

Comment:
This is going to break `#![crate_type]`.

Reply:
@0dvictor can you add a FIXME here to address this in the future?

Reply:
Done.